packages/test/test-end-to-end-tests/src/test/messageSize.spec.ts

```diff
@@ -144,4 +145,33 @@ describeNoCompat("Message size", (getTestObjectProvider) => {
 
         assertMapValues(dataObject2map, messageCount, largeString);
     });
+
+    it("Single large op passes when compression enabled and over max op size", async () => {
+        const maxMessageSizeInBytes = 1024 * 1024; // 1Mb
```

Comment:
@justus-camp The tests are failing because the runtime is erroring out due to the batch being too large _before_ compression actually happens. In light of this being experimental and also already having another layer which would catch this error (see the maxConsecutiveReconnects) I think it would be safe for now for the BatchManager to carry on adding stuff to its internal storage when the batch size exceeds any limits iff compression is enabled and the size exceeds the min compression size config.

Reply:
Yep, that's the strategy I'm currently implementing and about to push. I wish there was some sort of incremental "streaming" compression we could do where we can validate the batch size while compressing, but I don't see that capability inside of the lz4 library we're using.